clients/repo.go

```diff
@@ -17,6 +17,7 @@ package clients
 // Repo interface uniquely identifies a repo.
 type Repo interface {
 	URI() string
+	Host() string
```

Comment:
We don't seem to use this `Host()` API much apart from in tests. Any reason to add this?

Reply:
The main reason is that GitLab enterprise hosts (and GitHub enterprise hosts) may vary from just `gitlab|github.com`, so exposing the parameter may help for logging, etc.